newIDE/app/src/EventsSheet/EventsTree/index.js

```diff
@@ -400,22 +431,44 @@ export default class ThemableEventsTree extends Component<
     this.forceEventsUpdate();
   }
 
-  _getEventRow(searchedEvent: gdBaseEvent) {
+  getEventRow(searchedEvent: gdBaseEvent) {
     // TODO: flatData could be replaced by a hashmap of events to row index
     return findIndex(
       this.state.flatData,
       event => event.ptr === searchedEvent.ptr
     );
   }
 
+  getEventContextAtRowIndexes(rowIndexes: Array<number>): Array<EventContext> {
+    const flatDataTree: Array<{ node: SortableTreeNode }> = getFlatDataFromTree(
+      {
+        treeData: this.state.treeData,
+        getNodeKey,
+        ignoreCollapsed: true,
+      }
+    );
+    return rowIndexes
+      .map(rowIndex => {
+        if (
+          !flatDataTree[rowIndex] ||
+          (this.state.flatData.length < rowIndex || rowIndex < 0)
```

Comment:
Those extra checks don't seem useful to me, `flatDataTree[-1]` and `flatDataTree[flatDataTree.length + 2]` both return `undefined`

Reply:
OR: I'm not sure why there is a check on `this.state.flatData` here even though work seems to be done on `flatDataTree`